### PR TITLE
Add AWS Cloudtrail event source plugin

### PR DESCRIPTION
**Pull request by abikouo**
This PR adds support to AWS Cloudtrail as an event source and includes an example rulebook.

### CODE REVIEW

File: plugins/event_source/aws_cloudtrail.py
Comment (jillr) [inline plugins/event_source/aws_cloudtrail.py:8]:
```suggestion
    connection - Parameters used to create AWS session
    lookup_attributes - List of attributes to filter the events returned. Each item of the list should contain an attribute key and value.
```

File: plugins/event_source/aws_cloudtrail.py
Comment (jillr) [inline plugins/event_source/aws_cloudtrail.py:22]: What do you think about having `lookup_attributes` take a mapping of key:value pairs and convert them in the plugin to the format that boto requires?  Similar to `amazon.aws.module_utils.transofrmation.ansible_dict_to_boto3_filter_list` or  `amazon.aws.module_utils.tagging.ansible_dict_to_boto3_tag_list`, like this:

```
        lookup_attributes:
            eventSource: ec2.amazonaws.com
```

Comment (abikouo) [reply plugins/event_source/aws_cloudtrail.py:22]: Good idea, I will do that

**Review by benthomasson — DISMISSED** [review]

**Review by jillr — APPROVED** [review]
LGTM after a rebase

**Comment by abikouo** [comment]
> LGTM after a rebase

@jillr pull request has been rebased
Thanks

**Review by jillr — APPROVED** [review]

File: plugins/event_source/aws_cloudtrail.py
Comment (bzwei) [inline plugins/event_source/aws_cloudtrail.py:80]: Seems you should use aiobotocore.session. Take a look at similar implementation at https://github.com/ansible/event-driven-ansible/blob/main/plugins/event_source/sqs.py

File: plugins/event_source/aws_cloudtrail.py
Comment (abikouo) [reply plugins/event_source/aws_cloudtrail.py:8]: updated

File: plugins/event_source/aws_cloudtrail.py
Comment (abikouo) [reply plugins/event_source/aws_cloudtrail.py:44]: I will revert this and keep filtering as specified in the API

File: plugins/event_source/aws_cloudtrail.py
Comment (bzwei) [inline plugins/event_source/aws_cloudtrail.py:7]: can you added supported parameters? https://aiobotocore.readthedocs.io/en/latest/tutorial.html

File: plugins/event_source/aws_cloudtrail.py
Comment (bzwei) [inline plugins/event_source/aws_cloudtrail.py:100]: Is https://github.com/ansible/event-driven-ansible/blob/main/plugins/event_source/sqs.py#L46 only for sqs service or for all aws services? This way we don't need to do polling.

File: requirements.txt
Comment (abikouo) [reply requirements.txt:7]: yes sure

File: plugins/event_source/aws_cloudtrail.py
Comment (abikouo) [reply plugins/event_source/aws_cloudtrail.py:26]: mistake

File: plugins/event_source/aws_cloudtrail.py
Comment (abikouo) [reply plugins/event_source/aws_cloudtrail.py:100]: I am not aware of this for the cloudtrail event, I need to investigate if we can use something similar and remove user polling

**Comment by abikouo** [comment]
@jillr @bzwei do you know who can merge this?

**Comment by bzwei** [comment]
> @jillr @bzwei do you know who can merge this?

Need one more approval

**Comment by abikouo** [comment]
@jillr @benthomasson could you please review/approve once again? Thanks

**Review by benthomasson — APPROVED** [review]

**Pull request merged**
